### ISSUE TITLE
Broadcast of Intent ANR

### ISSUE THREAD

**Issue by needz** [google/ExoPlayer#4362]
### Issue description
Since ExoPlayer version 2.8.* I am receiving a lot of ANRs in the Google Play Console.
`Broadcast of Intent { act=android.intent.action.TIME_TICK flg=0x50000014 (has extras) }`
Seems to have something in common with https://github.com/google/ExoPlayer/issues/4352 , because there are ANRs when releasing the player.

### Reproduction steps
I cannot reproduce it, seems like it depends on the type of stream.

### Version of ExoPlayer being used
ExoPlayer 2.8.1

### Device(s) and version(s) of Android being used
Sony BRAVIA 4K GB (BRAVIA_ATV2), 2560MB RAM, Android 7.0
Bouygues Telecom Bbox Miami (HMB4213H), 2048MB RAM, Android 5.1
Xiaomi Mi Box (once), 2048MB RAM, Android 6.0
TIM TIMvision (dwt765ti), 2048MB RAM, Android 7.0
Philips 4K LED TV powered by Android (PH7M_EU_5596), 2560MB RAM, Android 7.0
etc.

### A full bug report captured from the device

```
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x741a8668 self=0xa9f85400
  | sysTid=722 nice=0 cgrp=default sched=0/0 handle=0xacc19534
  | state=S schedstat=( 4495701312 1963517425 12473 ) utm=381 stm=68 core=3 HZ=100
  | stack=0xbe03b000-0xbe03d000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0443c7f6> (a com.google.android.exoplayer2.z)
  at com.google.android.exoplayer2.PlayerMessage.blockUntilDelivered (PlayerMessage.java:282)
  at com.google.android.exoplayer2.SimpleExoPlayer.setVideoSurfaceInternal (SimpleExoPlayer.java:962)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$102 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$200 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$302 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$802 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$1002 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer.access$1300 (SimpleExoPlayer.java:59)
  at com.google.android.exoplayer2.SimpleExoPlayer$ComponentListener.surfaceDestroyed (SimpleExoPlayer.java:1134)
  at android.view.SurfaceView.updateWindow (SurfaceView.java:592)
  at android.view.SurfaceView.onWindowVisibilityChanged (SurfaceView.java:251)
  at android.view.View.dispatchWindowVisibilityChanged (View.java:10262)
  at android.view.ViewGroup.dispatchWindowVisibilityChanged (ViewGroup.java:1289)
  at android.view.ViewGroup.dispatchWindowVisibilityChanged (ViewGroup.java:1289)
  at android.view.ViewGroup.dispatchWindowVisibilityChanged (ViewGroup.java:1289)
  at android.view.ViewRootImpl.performTraversals (ViewRootImpl.java:1536)
  at android.view.ViewRootImpl.doTraversal (ViewRootImpl.java:1248)
  at android.view.ViewRootImpl$TraversalRunnable.run (ViewRootImpl.java:6312)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:871)
  at android.view.Choreographer.doCallbacks (Choreographer.java:683)
  at android.view.Choreographer.doFrame (Choreographer.java:619)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:857)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6077)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:865)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:755)
"Picasso-Stats" tid=16 Native 
"Picasso-Stats" prio=5 tid=16 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13670 self=0xa27d9800
  | sysTid=767 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x8d1fa920
  | state=S schedstat=( 9773007 72453599 128 ) utm=0 stm=0 core=3 HZ=100
  | stack=0x8d0f8000-0x8d0fa000 stackSize=1038KB
  | held mutexes=
  #00  pc 0000000000048388  /system/lib/libc.so (__epoll_pwait+20)
  #01  pc 000000000001a08d  /system/lib/libc.so (epoll_pwait+60)
  #02  pc 000000000001a0bd  /system/lib/libc.so (epoll_wait+12)
  #03  pc 0000000000011d8f  /system/lib/libutils.so (_ZN7android6Looper9pollInnerEi+118)
  #04  pc 0000000000011c8b  /system/lib/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+26)
  #05  pc 000000000008facd  /system/lib/libandroid_runtime.so (_ZN7android18NativeMessageQueue8pollOnceEP7_JNIEnvP8_jobjecti+22)
  #06  pc 00000000005f48e5  /data/dalvik-cache/arm/system@framework@boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+96)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:323)
  at android.os.Looper.loop (Looper.java:136)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"Picasso-Dispatcher" tid=17 Native 
"Picasso-Dispatcher" prio=5 tid=17 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13700 self=0xa27d9d00
  | sysTid=769 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x8ce7f920
  | state=S schedstat=( 13371502 21591541 131 ) utm=0 stm=1 core=3 HZ=100
  | stack=0x8cd7d000-0x8cd7f000 stackSize=1038KB
  | held mutexes=
  #00  pc 0000000000048388  /system/lib/libc.so (__epoll_pwait+20)
  #01  pc 000000000001a08d  /system/lib/libc.so (epoll_pwait+60)
  #02  pc 000000000001a0bd  /system/lib/libc.so (epoll_wait+12)
  #03  pc 0000000000011d8f  /system/lib/libutils.so (_ZN7android6Looper9pollInnerEi+118)
  #04  pc 0000000000011c8b  /system/lib/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+26)
  #05  pc 000000000008facd  /system/lib/libandroid_runtime.so (_ZN7android18NativeMessageQueue8pollOnceEP7_JNIEnvP8_jobjecti+22)
  #06  pc 00000000005f48e5  /data/dalvik-cache/arm/system@framework@boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+96)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:323)
  at android.os.Looper.loop (Looper.java:136)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"Jit thread pool worker thread 0" tid=2 Native 
"Jit thread pool worker thread 0" prio=5 tid=2 Native (still starting up)
  | group="" sCount=1 dsCount=0 obj=0x0 self=0xa258c000
  | sysTid=727 nice=9 cgrp=default sched=0/0 handle=0xa96aa920
  | state=S schedstat=( 1265459 936167 7 ) utm=0 stm=0 core=1 HZ=100
  | stack=0xa95ac000-0xa95ae000 stackSize=1022KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 00000000000b633d  /system/lib/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+92)
  #02  pc 00000000003404f9  /system/lib/libart.so (_ZN3art10ThreadPool7GetTaskEPNS_6ThreadE+160)
  #03  pc 000000000033fd23  /system/lib/libart.so (_ZN3art16ThreadPoolWorker3RunEv+62)
  #04  pc 000000000033f851  /system/lib/libart.so (_ZN3art16ThreadPoolWorker8CallbackEPv+64)
  #05  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #06  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"ReferenceQueueDaemon" tid=4 Waiting 
"ReferenceQueueDaemon" daemon prio=5 tid=4 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x32c13160 self=0x9fb11100
  | sysTid=729 nice=0 cgrp=default sched=0/0 handle=0xa94aa920
  | state=S schedstat=( 2069582 3462920 32 ) utm=0 stm=0 core=0 HZ=100
  | stack=0xa93a8000-0xa93aa000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x060d47f7> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
  at java.lang.Daemons$ReferenceQueueDaemon.run (Daemons.java:150)
- locked <0x060d47f7> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
  at java.lang.Thread.run (Thread.java:761)
"FinalizerDaemon" tid=5 Waiting 
"FinalizerDaemon" daemon prio=5 tid=5 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x32c131f0 self=0x9fb11600
  | sysTid=730 nice=0 cgrp=default sched=0/0 handle=0xa93a5920
  | state=S schedstat=( 4065624 3195751 28 ) utm=0 stm=0 core=0 HZ=100
  | stack=0xa92a3000-0xa92a5000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0f9f1664> (a java.lang.Object)
  at java.lang.Object.wait (Object.java:407)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:188)
- locked <0x0f9f1664> (a java.lang.Object)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:209)
  at java.lang.Daemons$FinalizerDaemon.run (Daemons.java:204)
  at java.lang.Thread.run (Thread.java:761)
"FinalizerWatchdogDaemon" tid=6 Waiting 
"FinalizerWatchdogDaemon" daemon prio=5 tid=6 Waiting
  | group="system" sCount=1 dsCount=0 obj=0x32c13280 self=0x9fb11b00
  | sysTid=731 nice=0 cgrp=default sched=0/0 handle=0xa92a0920
  | state=S schedstat=( 606416 1468541 13 ) utm=0 stm=0 core=2 HZ=100
  | stack=0xa919e000-0xa91a0000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x04e730cd> (a java.lang.Daemons$FinalizerWatchdogDaemon)
  at java.lang.Daemons$FinalizerWatchdogDaemon.sleepUntilNeeded (Daemons.java:269)
- locked <0x04e730cd> (a java.lang.Daemons$FinalizerWatchdogDaemon)
  at java.lang.Daemons$FinalizerWatchdogDaemon.run (Daemons.java:249)
  at java.lang.Thread.run (Thread.java:761)
"Binder:722_1" tid=8 Native 
"Binder:722_1" prio=5 tid=8 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c133a0 self=0xa258d400
  | sysTid=733 nice=0 cgrp=default sched=0/0 handle=0xa8f98920
  | state=S schedstat=( 229369994 623351925 3676 ) utm=6 stm=16 core=0 HZ=100
  | stack=0xa8e9c000-0xa8e9e000 stackSize=1014KB
  | held mutexes=
  #00  pc 00000000000484c8  /system/lib/libc.so (__ioctl+8)
  #01  pc 000000000001acdf  /system/lib/libc.so (ioctl+38)
  #02  pc 000000000003cc55  /system/lib/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+168)
  #03  pc 000000000003cd4d  /system/lib/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+8)
  #04  pc 000000000003d2bb  /system/lib/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+46)
  #05  pc 000000000004f5f5  /system/lib/libbinder.so (???)
  #06  pc 000000000000e3a1  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+140)
  #07  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #08  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #09  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Binder:722_2" tid=9 Native 
"Binder:722_2" prio=5 tid=9 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13430 self=0xa9f86800
  | sysTid=734 nice=0 cgrp=default sched=0/0 handle=0xa8e99920
  | state=S schedstat=( 230189141 615918178 3684 ) utm=5 stm=18 core=2 HZ=100
  | stack=0xa8d9d000-0xa8d9f000 stackSize=1014KB
  | held mutexes=
  #00  pc 00000000000484c8  /system/lib/libc.so (__ioctl+8)
  #01  pc 000000000001acdf  /system/lib/libc.so (ioctl+38)
  #02  pc 000000000003cc55  /system/lib/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+168)
  #03  pc 000000000003cd4d  /system/lib/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+8)
  #04  pc 000000000003d2bb  /system/lib/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+46)
  #05  pc 000000000004f5f5  /system/lib/libbinder.so (???)
  #06  pc 000000000000e3a1  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+140)
  #07  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #08  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #09  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Profile Saver" tid=10 Native 
"Profile Saver" daemon prio=5 tid=10 Native
  | group="system" sCount=1 dsCount=0 obj=0x32c134c0 self=0xa258e800
  | sysTid=736 nice=0 cgrp=default sched=0/0 handle=0xa8a6b920
  | state=S schedstat=( 9879834 2907333 23 ) utm=0 stm=0 core=2 HZ=100
  | stack=0xa896f000-0xa8971000 stackSize=1014KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 00000000000b633d  /system/lib/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+92)
  #02  pc 00000000002573c5  /system/lib/libart.so (_ZN3art12ProfileSaver3RunEv+296)
  #03  pc 0000000000258701  /system/lib/libart.so (_ZN3art12ProfileSaver21RunProfileSaverThreadEPv+52)
  #04  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #05  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Binder:722_3" tid=12 Native 
"Binder:722_3" prio=5 tid=12 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13d30 self=0x8cf89c00
  | sysTid=989 nice=0 cgrp=default sched=0/0 handle=0xa257f920
  | state=S schedstat=( 54722962 118003469 747 ) utm=1 stm=3 core=3 HZ=100
  | stack=0xa2483000-0xa2485000 stackSize=1014KB
  | held mutexes=
  #00  pc 00000000000484c8  /system/lib/libc.so (__ioctl+8)
  #01  pc 000000000001acdf  /system/lib/libc.so (ioctl+38)
  #02  pc 000000000003cc55  /system/lib/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+168)
  #03  pc 000000000003cd4d  /system/lib/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+8)
  #04  pc 000000000003d2bb  /system/lib/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+46)
  #05  pc 000000000004f5f5  /system/lib/libbinder.so (???)
  #06  pc 000000000000e3a1  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+140)
  #07  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #08  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #09  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"RenderThread" tid=13 Native 
"RenderThread" prio=5 tid=13 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13550 self=0x9fab1100
  | sysTid=748 nice=-4 cgrp=default sched=0/0 handle=0xa247a920
  | state=S schedstat=( 14033802451 2395452471 34565 ) utm=1046 stm=357 core=0 HZ=100
  | stack=0xa237e000-0xa2380000 stackSize=1014KB
  | held mutexes=
  #00  pc 0000000000048388  /system/lib/libc.so (__epoll_pwait+20)
  #01  pc 000000000001a08d  /system/lib/libc.so (epoll_pwait+60)
  #02  pc 000000000001a0bd  /system/lib/libc.so (epoll_wait+12)
  #03  pc 0000000000011d8f  /system/lib/libutils.so (_ZN7android6Looper9pollInnerEi+118)
  #04  pc 0000000000011c8b  /system/lib/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+26)
  #05  pc 0000000000028393  /system/lib/libhwui.so (_ZN7android10uirenderer12renderthread12RenderThread10threadLoopEv+382)
  #06  pc 000000000000e3a1  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+140)
  #07  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #08  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #09  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"hwuiTask1" tid=15 Native 
"hwuiTask1" prio=5 tid=15 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c135e0 self=0xa258fc00
  | sysTid=764 nice=-2 cgrp=default sched=0/0 handle=0x8dd4b920
  | state=S schedstat=( 286550328 435750085 4327 ) utm=24 stm=4 core=2 HZ=100
  | stack=0x8dc4f000-0x8dc51000 stackSize=1014KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 0000000000046a9d  /system/lib/libc.so (_ZL24__pthread_cond_timedwaitP23pthread_cond_internal_tP15pthread_mutex_tbPK8timespec+102)
  #02  pc 0000000000028e03  /system/lib/libhwui.so (???)
  #03  pc 0000000000028d79  /system/lib/libhwui.so (???)
  #04  pc 000000000000e41d  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+264)
  #05  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #06  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #07  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Picasso-refQueue" tid=18 TimedWaiting 
"Picasso-refQueue" daemon prio=5 tid=18 TimedWaiting
  | group="main" sCount=1 dsCount=0 obj=0x32c13790 self=0xa27da200
  | sysTid=770 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x8ccae920
  | state=S schedstat=( 11471580 42678298 125 ) utm=0 stm=1 core=1 HZ=100
  | stack=0x8cbac000-0x8cbae000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x02dc6d82> (a java.lang.Object)
  at java.lang.Object.wait (Object.java:407)
  at java.lang.ref.ReferenceQueue.remove (ReferenceQueue.java:188)
- locked <0x02dc6d82> (a java.lang.Object)
  at com.squareup.picasso.t$b.run (Picasso.java:631)
"Picasso-Idle" tid=19 Waiting 
"Picasso-Idle" prio=5 tid=19 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x32c13820 self=0x8cf86000
  | sysTid=771 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x8cb23920
  | state=S schedstat=( 3491167 908709 6 ) utm=0 stm=0 core=2 HZ=100
  | stack=0x8ca21000-0x8ca23000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x05c46c93> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2127)
- locked <0x05c46c93> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:161)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2035)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:519)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1058)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1118)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:761)
  at com.squareup.picasso.ad$a.run (Utils.java:354)
"Picasso-Idle" tid=20 Waiting 
"Picasso-Idle" prio=5 tid=20 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x32c138b0 self=0x8cf86500
  | sysTid=773 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x8ca1e920
  | state=S schedstat=( 689125 432667 1 ) utm=0 stm=0 core=0 HZ=100
  | stack=0x8c91c000-0x8c91e000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0a6934d0> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2127)
- locked <0x0a6934d0> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:325)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:161)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await (AbstractQueuedSynchronizer.java:2035)
  at java.util.concurrent.PriorityBlockingQueue.take (PriorityBlockingQueue.java:519)
  at java.util.concurrent.ThreadPoolExecutor.getTask (ThreadPoolExecutor.java:1058)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1118)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:761)
  at com.squareup.picasso.ad$a.run (Utils.java:354)
"ChoreographerOwner:Handler" tid=21 Native 
"ChoreographerOwner:Handler" prio=5 tid=21 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13940 self=0xa27da700
  | sysTid=774 nice=0 cgrp=default sched=0/0 handle=0x8c919920
  | state=S schedstat=( 86728597 151996604 492 ) utm=6 stm=2 core=3 HZ=100
  | stack=0x8c817000-0x8c819000 stackSize=1038KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 00000000000b633d  /system/lib/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+92)
  #02  pc 00000000003ef76b  /system/lib/libart.so (_ZN3artL12GoToRunnableEPNS_6ThreadE+230)
  #03  pc 00000000003ef65b  /system/lib/libart.so (_ZN3art12JniMethodEndEjPNS_6ThreadE+8)
  #04  pc 00000000005f48ef  /data/dalvik-cache/arm/system@framework@boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+106)
  at android.os.MessageQueue.nativePollOnce (Native method)
  at android.os.MessageQueue.next (MessageQueue.java:323)
  at android.os.Looper.loop (Looper.java:136)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"ExoPlayerImplInternal:Handler" tid=22 Native 
"ExoPlayerImplInternal:Handler" prio=5 tid=22 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c139d0 self=0xa27dac00
  | sysTid=775 nice=-16 cgrp=default sched=0/0 handle=0x8c814920
  | state=S schedstat=( 254644151 42125707 794 ) utm=19 stm=6 core=1 HZ=100
  | stack=0x8c712000-0x8c714000 stackSize=1038KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 0000000000046a9d  /system/lib/libc.so (_ZL24__pthread_cond_timedwaitP23pthread_cond_internal_tP15pthread_mutex_tbPK8timespec+102)
  #02  pc 000000000000f8c9  /system/lib/libstagefright_foundation.so (_ZN7android7ALooper13awaitResponseERKNS_2spINS_11AReplyTokenEEEPNS1_INS_8AMessageEEE+92)
  #03  pc 0000000000010d6f  /system/lib/libstagefright_foundation.so (_ZN7android8AMessage20postAndAwaitResponseEPNS_2spIS0_EE+138)
  #04  pc 00000000000bfa25  /system/lib/libstagefright.so (_ZN7android10MediaCodec20PostAndAwaitResponseERKNS_2spINS_8AMessageEEEPS3_+20)
  #05  pc 00000000000bf02d  /system/lib/libstagefright.so (_ZN7android10MediaCodec4initERKNS_7AStringEbb+964)
  #06  pc 00000000000bf10f  /system/lib/libstagefright.so (_ZN7android10MediaCodec21CreateByComponentNameERKNS_2spINS_7ALooperEEERKNS_7AStringEPii+54)
  #07  pc 000000000001f989  /system/lib/libmedia_jni.so (_ZN7android11JMediaCodecC1EP7_JNIEnvP8_jobjectPKcbb+300)
  #08  pc 000000000002310b  /system/lib/libmedia_jni.so (???)
  #09  pc 00000000004373f7  /data/dalvik-cache/arm/system@framework@boot-framework.oat (Java_android_media_MediaCodec_native_1setup__Ljava_lang_String_2ZZ+114)
  at android.media.MediaCodec.native_setup (Native method)
  at android.media.MediaCodec.<init> (MediaCodec.java:1774)
  at android.media.MediaCodec.createByCodecName (MediaCodec.java:1755)
  at com.google.android.exoplayer2.f.b.y (MediaCodecRenderer.java:415)
  at com.google.android.exoplayer2.f.b.b (MediaCodecRenderer.java:920)
  at com.google.android.exoplayer2.m.e.b (MediaCodecVideoRenderer.java:503)
  at com.google.android.exoplayer2.f.b.a (MediaCodecRenderer.java:557)
  at com.google.android.exoplayer2.l.h (ExoPlayerImplInternal.java:518)
  at com.google.android.exoplayer2.l.handleMessage (ExoPlayerImplInternal.java:301)
  at android.os.Handler.dispatchMessage (Handler.java:98)
  at android.os.Looper.loop (Looper.java:154)
  at android.os.HandlerThread.run (HandlerThread.java:61)
"hwuiTask2" tid=23 Native 
"hwuiTask2" prio=5 tid=23 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13af0 self=0x8cf87400
  | sysTid=776 nice=-2 cgrp=default sched=0/0 handle=0x8acff920
  | state=S schedstat=( 642250 898416 9 ) utm=0 stm=0 core=2 HZ=100
  | stack=0x8ac03000-0x8ac05000 stackSize=1014KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 0000000000046a9d  /system/lib/libc.so (_ZL24__pthread_cond_timedwaitP23pthread_cond_internal_tP15pthread_mutex_tbPK8timespec+102)
  #02  pc 0000000000028e03  /system/lib/libhwui.so (???)
  #03  pc 0000000000028d79  /system/lib/libhwui.so (???)
  #04  pc 000000000000e41d  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+264)
  #05  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #06  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #07  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Loader:ExtractorMediaPeriod" tid=24 Waiting 
"Loader:ExtractorMediaPeriod" prio=5 tid=24 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x32c13b80 self=0x8ad04900
  | sysTid=777 nice=0 cgrp=default sched=0/0 handle=0x8ac00920
  | state=S schedstat=( 200018018 149615400 249 ) utm=17 stm=3 core=2 HZ=100
  | stack=0x8aafe000-0x8ab00000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x05c30cc9> (a com.google.android.exoplayer2.l.e)
  at com.google.android.exoplayer2.l.e.c (ConditionVariable.java:58)
- locked <0x05c30cc9> (a com.google.android.exoplayer2.l.e)
  at com.google.android.exoplayer2.h.h$a.c (ExtractorMediaPeriod.java:859)
  at com.google.android.exoplayer2.k.q$b.run (Loader.java:317)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1133)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:761)
"OkHttp ConnectionPool" tid=26 TimedWaiting 
"OkHttp ConnectionPool" daemon prio=5 tid=26 TimedWaiting
  | group="main" sCount=1 dsCount=0 obj=0x32c13c10 self=0x8ad03000
  | sysTid=860 nice=0 cgrp=default sched=0/0 handle=0x8a7ff920
  | state=S schedstat=( 826417 406250 3 ) utm=0 stm=0 core=0 HZ=100
  | stack=0x8a6fd000-0x8a6ff000 stackSize=1038KB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0e67dfce> (a okhttp3.j)
  at okhttp3.j$1.run (ConnectionPool.java:67)
- locked <0x0e67dfce> (a okhttp3.j)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1133)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:761)
"MediaCodec_looper" tid=28 Native 
"MediaCodec_looper" prio=6 tid=28 Native
  | group="main" sCount=1 dsCount=0 obj=0x32c13ca0 self=0x8cf88800
  | sysTid=885 nice=-2 cgrp=default sched=0/0 handle=0x8dfc7920
  | state=S schedstat=( 1370542 2994457 16 ) utm=0 stm=0 core=1 HZ=100
  | stack=0x8decb000-0x8decd000 stackSize=1014KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 0000000000046a9d  /system/lib/libc.so (_ZL24__pthread_cond_timedwaitP23pthread_cond_internal_tP15pthread_mutex_tbPK8timespec+102)
  #02  pc 000000000000f53b  /system/lib/libstagefright_foundation.so (_ZN7android7ALooper4loopEv+298)
  #03  pc 000000000000e41d  /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+264)
  #04  pc 0000000000064ea1  /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+80)
  #05  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #06  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"Signal Catcher" tid=3 Runnable 
"Signal Catcher" daemon prio=5 tid=3 Runnable
  | group="system" sCount=0 dsCount=0 obj=0x32c130d0 self=0x9fb0f800
  | sysTid=728 nice=0 cgrp=default sched=0/0 handle=0xa95a9920
  | state=R schedstat=( 135883608 181058432 991 ) utm=2 stm=11 core=1 HZ=100
  | stack=0xa94ad000-0xa94af000 stackSize=1014KB
  | held mutexes= "mutator lock"(shared held)
  #00  pc 000000000034c795  /system/lib/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+128)
  #01  pc 000000000032d151  /system/lib/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMap+308)
  #02  pc 000000000033efb1  /system/lib/libart.so (_ZN3art14DumpCheckpoint3RunEPNS_6ThreadE+620)
  #03  pc 0000000000338fbb  /system/lib/libart.so (_ZN3art10ThreadList13RunCheckpointEPNS_7ClosureE+330)
  #04  pc 0000000000338ce1  /system/lib/libart.so (_ZN3art10ThreadList4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEb+180)
  #05  pc 0000000000338bcf  /system/lib/libart.so (_ZN3art10ThreadList14DumpForSigQuitERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+574)
  #06  pc 000000000031e27d  /system/lib/libart.so (_ZN3art7Runtime14DumpForSigQuitERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+228)
  #07  pc 0000000000322ae7  /system/lib/libart.so (_ZN3art13SignalCatcher13HandleSigQuitEv+1394)
  #08  pc 0000000000321c55  /system/lib/libart.so (_ZN3art13SignalCatcher3RunEPv+336)
  #09  pc 0000000000046f13  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  #10  pc 0000000000019cdd  /system/lib/libc.so (__start_thread+6)
"HeapTaskDaemon" tid=7 Blocked 
"HeapTaskDaemon" daemon prio=5 tid=7 Blocked
  | group="system" sCount=1 dsCount=0 obj=0x32c13310 self=0xa9f85e00
  | sysTid=732 nice=0 cgrp=default sched=0/0 handle=0xa919b920
  | state=S schedstat=( 405548524 191650557 376 ) utm=33 stm=7 core=0 HZ=100
  | stack=0xa9099000-0xa909b000 stackSize=1038KB
  | held mutexes=
  #00  pc 0000000000017404  /system/lib/libc.so (syscall+28)
  #01  pc 00000000000b633d  /system/lib/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+92)
  #02  pc 00000000001b3465  /system/lib/libart.so (_ZN3art2gc13TaskProcessor7GetTaskEPNS_6ThreadE+288)
  #03  pc 00000000001b3add  /system/lib/libart.so (_ZN3art2gc13TaskProcessor11RunAllTasksEPNS_6ThreadE+44)
  #04  pc 0000000000157357  /data/dalvik-cache/arm/system@framework@boot-core-libart.oat (Java_dalvik_system_VMRuntime_runHeapTasks__+74)
  at dalvik.system.VMRuntime.runHeapTasks (Native method)
- waiting to lock an unknown object
  at java.lang.Daemons$HeapTaskDaemon.run (Daemons.java:433)
  at java.lang.Thread.run (Thread.java:761)
```

**Comment by ojw28** [comment]
Let's mark this as a duplicate of #4352.
